project/SparkSnowflakeBuild.scala

```diff
@@ -45,10 +45,10 @@ object SparkSnowflakeBuild extends Build {
       organization := "net.snowflake",
       scalaVersion := sys.props.getOrElse("SPARK_SCALA_VERSION", default = "2.11.7"),
       crossScalaVersions := Seq("2.10.5", "2.11.7"),
-      sparkVersion := "2.1.0",
+      sparkVersion := "2.2.0",
       testSparkVersion := sys.props.get("spark.testVersion").getOrElse(sparkVersion.value),
-      testHadoopVersion := sys.props.get("hadoop.testVersion").getOrElse("2.2.0"),
-      javacOptions ++= Seq("-source", "1.6", "-target", "1.6"),
+      testHadoopVersion := sys.props.get("hadoop.testVersion").getOrElse("2.6.0"),
+      javacOptions ++= Seq("-source", "1.8", "-target", "1.8"),
```

Comment:
Let's try to keep Java 1.6 if possible (for backwards compatibility reasons). Our current dependency is Java 6; we had problems with customers even when we tried to increase this to Java 7.

Reply:
really? Java 1.6 reached end of life in 2013... I'd looked into making equivalent changes locally and decided it wasn't worth the effort of PR'ing/etc given the legacy versions of many things...

Per [Spark documentation](https://spark.apache.org/docs/2.2.0/): 

```Note that support for Java 7, Python 2.6 and old Hadoop versions before 2.6.5 were removed as of Spark 2.2.0.```

If you need to maintain support for customers who are relying on legacy versions, then perhaps consider maintaining 2 versions of this connector -- the legacy one, and a version that is current with the ecosystem.